library/Reader.php

```diff
@@ -89,6 +89,8 @@ private function init() {
         if ($version!=self::FILE_FORMAT_VERSION)
             throw new Exception('Datafile not correct version. Found '.$version.' expected '.self::FILE_FORMAT_VERSION);
         $this->functionPos = $this->read($functionCount);
+        if (!is_array($this->functionPos))
```

Comment:
I think the logic here would be more quickly understood if you used something like `$functionCount == 1` as the conditional.

Reply:
I was thinking the same but decided to use the is_array so won't depend on how read function works internally and use the return function signature type that can be unsigned long or array